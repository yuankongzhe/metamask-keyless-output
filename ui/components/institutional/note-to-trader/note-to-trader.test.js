import { render, fireEvent } from '@testing-library/react';
import React from 'react';
import sinon from 'sinon';
import NoteToTrader from './note-to-trader';

describe('NoteToTrader', () => {
  it('should render the Note to trader component', () => {
    const props = {
      placeholder: '',
      maxLength: '280',
      noteText: 'some text',
      labelText: 'Transaction note',
      onChange: sinon.spy(),
    };

    const { getByTestId, container } = render(<NoteToTrader {...props} />);

    fireEvent.change(getByTestId('transaction-note'));
    expect(getByTestId('transaction-note').value).toBe('some text');
    expect(getByTestId('transaction-note')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
