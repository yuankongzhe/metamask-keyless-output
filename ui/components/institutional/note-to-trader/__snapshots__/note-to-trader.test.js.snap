// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteToTrader should render the Note to trader component 1`] = `
<div>
  <div
    class="mm-box confirm-page-container-content__data"
  >
    <div
      class="mm-box mm-box--padding-4 mm-box--display-flex mm-box--flex-direction-column"
    >
      <div
        class="mm-box note-header mm-box--display-flex mm-box--justify-content-space-between"
      >
        <label
          class="mm-box mm-text mm-label mm-label--html-for mm-text--body-md mm-text--font-weight-medium mm-box--display-inline-flex mm-box--align-items-center mm-box--color-text-default"
          for="transaction-note"
        >
          Transaction note
        </label>
        <p
          class="mm-box mm-text note-header__counter mm-text--body-md mm-box--color-text-default"
        >
          9
          /
          280
        </p>
      </div>
      <div
        class="mm-box note-field mm-box--display-flex mm-box--flex-direction-column"
      >
        <textarea
          data-testid="transaction-note"
          id="transaction-note"
          maxlength="280"
          placeholder=""
        >
          some text
        </textarea>
      </div>
    </div>
  </div>
</div>
`;
